immunopepper/preprocess.py

```diff
@@ -363,190 +370,183 @@ def parse_gene_metadata_info(h5fname, sample_list, cross_graph_expr):
     return countinfo, matching_count_samples, matching_count_ids
 
 
-def parse_mutation_from_vcf(mutation_tag, vcf_path, output_dir='', heter_code=0, mut_pickle=False, target_sample_list=None, mutation_sample=None, sample_eq_dict={}):
-    """Extract germline mutation information from the given vcf file and vcf.h5 file
-
-    Parameters
-    ----------
-    vcf_path: str, vcf file path
-    output_dir: str, path to vcf pickle output directory
-    heter_code: int (0 or 2). specify which number represents heter alle.
+def parse_mutation_from_vcf(mutation_tag: str, vcf_path: str, output_dir: str = '', heter_code: int = 0,
+                            cache_result: bool = False,
+                            target_samples: list[str] = None, mutation_sample: str = None, sample_eq_dict={}):
+    """Extract mutation information from the given vcf or vcf.h5 file
+    :param mutation_mode: what mutations to apply to the reference.
+        One of 'ref', 'germline', 'somatic', 'somatic_and_germline'
+    :param maf_path: path to the VCF file to be read
+    :param output_dir: location where a pickle of the result can be cached if cache_result is true
+    :param heter_code: int (0 or 2). specify which number represents heter allele
         0: 0-> homozygous alternative(1|1), 1-> heterozygous(0|1,1|0) 2->homozygous reference(0|0)
         2: 0-> homozygous reference(0|0), 1-> heterozygous(0|1,1|0) 2->homozygous alternative(1|1)
-    mut_pickle: bool, flag indicating whether to pickle mutation info to disk
-    target_sample_list: list, list of samples
-
-    Returns
-    -------
-    mut_dict: with key (sample, chromo) and values (var_dict)
+    ::param cache_result: if True, attempt to load the data from a pre-created pickle file (to save time). If the pickle
+        file is not present, cache the result into a pickle file to be used the next run
+    :param target_samples: list of tissue sample ids TODO(dd) - what are these used for?
```

Comment:
This is related to this post https://ratschlab.slack.com/archives/C031DFX95S8/p1644833684099919?thread_ts=1644758830.505279&cid=C031DFX95S8